test/expect-tests/dune_console/dune_console_tests.ml

```diff
@@ -30,42 +27,107 @@ let test_status_line_clearing_with_wrapping (module Console : New_console) =
   let open Console in
   Status_line.set
     (Status_line.Constant
-       (Pp.text
-          "This status line is a problem because of the fact that it is especially long \
-           and therefore will not be cleared properly."));
+       (Pp.hovbox
+        @@ Pp.text
+             "This status line is a problem because of the fact that it is especially \
+              long and therefore will not be cleared properly."));
   Status_line.set (Status_line.Constant (Pp.text "Here is another"));
   Status_line.clear ()
 ;;
 
+module Escape = struct
```

Comment:
Why do we need this all this code? Why can't we just use `String.escaped` or something if it's just for tests?

Reply:
I don't want to escape all characters such as newlines and I don't see a way of making that modular.

Reply:
Why don't you want to escape the newlines?

Even if you don't, how about you split it by lines and print a list of lines? Or at least implement the escaping with `Re` to make it simple. You took a lot of code from our escaping functions that are bug prone and designed for performance. I don't want code like that in the test suite.

Reply:
I've implemented the escaping using Re now.

Reply:
I am purposefully not escaping newlines because I want them rendered in the test output. It makes comparing lines easier.

Reply:
Except you can do that with `String.split_on_char ~on:'\n' |> List.map ~f:String.escape` already.